extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,321 @@
+        keySelector,
+        valueSelectors,
+        nominalEntries,
+        getMaxIntermediateSize()
+    );
+  }
+
+  @Override
+  public Object deserialize(final Object object)
+  {
+    return ArrayOfDoublesSketchOperations.deserialize(object);
+  }
+
+  @Override
+  public Comparator<ArrayOfDoublesSketch> getComparator()
+  {
+    return COMPARATOR;
+  }
+
+  @Override
+  public Object combine(@Nullable final Object lhs, @Nullable final Object rhs)
+  {
+    final ArrayOfDoublesUnion union = new ArrayOfDoublesSetOperationBuilder().setNominalEntries(nominalEntries)
+        .setNumberOfValues(numberOfValues).buildUnion();
+    if (lhs != null) {
+      union.update((ArrayOfDoublesSketch) lhs);
+    }
+    if (rhs != null) {
+      union.update((ArrayOfDoublesSketch) rhs);
+    }
+    return union.getResult();
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @JsonProperty
+  public int getNominalEntries()
+  {
+    return nominalEntries;
+  }
+
+  @JsonProperty
+  public List<String> getMetricColumns()
+  {
+    return metricColumns;
+  }
+
+  @JsonProperty
+  public int getNumberOfValues()
+  {
+    return numberOfValues;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    return new CacheKeyBuilder(AggregatorUtil.ARRAY_OF_DOUBLES_SKETCH_CACHE_TYPE_ID)
+        .appendString(name)
+        .appendString(fieldName)
+        .appendInt(nominalEntries)
+        .appendStrings(metricColumns)
+        .appendInt(numberOfValues)
+        .build();
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return ArrayOfDoublesUnion.getMaxBytes(nominalEntries, numberOfValues);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.<AggregatorFactory> singletonList(
+      new ArrayOfDoublesSketchAggregatorFactory(
+          fieldName,
+          fieldName,
+          nominalEntries,
+          metricColumns,
+          numberOfValues
+      )
+    );
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new ArrayOfDoublesSketchAggregatorFactory(name, name, nominalEntries, null, numberOfValues);
+  }
+
+  @Override
+  public Object finalizeComputation(final Object object)
+  {
+    return ((ArrayOfDoublesSketch) object).getEstimate();
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    if (metricColumns == null) {
+      return ArrayOfDoublesSketchModule.ARRAY_OF_DOUBLES_SKETCH_MERGE_AGG;
+    }
+    return ArrayOfDoublesSketchModule.ARRAY_OF_DOUBLES_SKETCH_BUILD_AGG;
+  }
+
+  @Override
+  public boolean equals(final Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass().equals(o.getClass())) {
+      return false;
+    }
+    if (!super.equals(o)) {
```

Comment:
Not needed. AggregatorFactory is stateless (and it's soon better be made an interface, to avoid confusion)

Reply:
ok